shotover-proxy/src/transforms/protect/mod.rs

```diff
@@ -317,3 +387,131 @@ impl Transform for Protect {
         Ok(result)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::frame::CQL;
+    use crate::message::MessageValue;
+    use crate::transforms::protect::key_management::KeyManager;
+    use crate::transforms::protect::local_kek::LocalKeyManagement;
+    use crate::transforms::protect::{Protect, Protected};
+    use bytes::Bytes;
+    use cql3_parser::cassandra_statement::CassandraStatement;
+    use cql3_parser::common::Operand;
+    use cql3_parser::insert::InsertValues;
+    use sodiumoxide::crypto::secretbox::Nonce;
+    use std::collections::HashMap;
+
+    #[test]
+    fn test_serde() {
+        let n: [u8; 24] = [
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24,
+        ];
+        let ocipher = Bytes::from("this would be encrypted data").to_vec();
+        let ononce = Nonce::from_slice(&n).unwrap();
+        let oenc_dek = Bytes::from("this would be enc_dek").to_vec();
+        let okek_id = "The KEK id".to_string();
+
+        let protected = Protected::Ciphertext {
+            cipher: ocipher.clone(),
+            nonce: ononce,
+            enc_dek: oenc_dek.clone(),
+            kek_id: okek_id.clone(),
+        };
+        let encoded = Protect::encode(&protected).unwrap();
+        let decoded = Protect::decode(&encoded).unwrap();
+        if let Protected::Ciphertext {
+            cipher,
+            nonce,
+            enc_dek,
+            kek_id,
+        } = decoded
+        {
+            assert_eq!(&ocipher, &cipher);
+            assert_eq!(&ononce, &nonce);
+            assert_eq!(&oenc_dek, &enc_dek);
+            assert_eq!(&okek_id, &kek_id);
+        } else {
+            panic!("not a Ciphertext")
+        }
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    //#[test]
+    async fn round_trip_test() {
+        if sodiumoxide::init().is_err() {
```

Comment:
`sodiumoxide::init().expect("could not init sodiumoxide");` would at least include the actual error returned by init in the panic message